File: auth-jwt/auth-jwt-java-sbt/auth-jwt-java-sbt-api/src/main/java/com/example/authjwt/api/AuthJwtService.java
package com.example.authjwt.api;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

/**
 * Simple service for demonstration authenticate/authorize
 */
public interface AuthJwtService extends Service {

    /**
     * Authenticate
     */
    ServiceCall<NotUsed, String> headerJwtAuthenticate();

    /**
     * Authorize by role 'manager'
     */
    ServiceCall<NotUsed, String> headerJwtAuthorize();

    @Override
    default Descriptor descriptor() {
        return named("auth-jwt")
                .withCalls(
                        pathCall("/authenticate", this::headerJwtAuthenticate),
                        pathCall("/authorize", this::headerJwtAuthorize)
                )
                .withAutoAcl(true);
    }
}

File: auth-jwt/auth-jwt-scala-sbt/auth-jwt-scala-sbt-api/src/main/scala/com/example/authjwt/api/AuthJwtService.scala
package com.example.authjwt.api

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.{Service, ServiceCall}

/**
  * Simple service for demonstration authenticate/authorize
  */
trait AuthJwtService extends Service {

  /**
    * Authenticate
    */
  def headerJwtAuthenticate: ServiceCall[NotUsed, String]

  /**
    * Authorize by role 'manager'
    */
  def headerJwtAuthorize: ServiceCall[NotUsed, String]

  override final def descriptor = {
    import Service._
    // @formatter:off
    named("auth-jwt")
      .withCalls(
        pathCall("/authenticate", this.headerJwtAuthenticate),
        pathCall("/authorize", this.headerJwtAuthorize)
      )
      .withAutoAcl(true)
    // @formatter:on
  }
}
